### DISCUSSION TITLE
Compiling librespot on Android in Termux

### DISCUSSION THREAD

**Discussion by nmlt** [librespot-org/librespot#915] [General]
Hi,
I'm trying to use librespot in Termux on Android. After upgrading to the newest packages, and installing rust via the pkg apt wrapper, I'm getting the following error for running `cargo install librespot`:

```
.
.
.
Compiling jni v0.19.0
error[E0425]: cannot find value `EXPECTED_JVM_FILENAME` in this scope
   --> /data/data/com.termux/files/home/.cargo/registry/src/github.com-1285ae84e5963aae/jni-0.19.0/build.rs:102:25
    |
102 |         if file_name == EXPECTED_JVM_FILENAME {
    |                         ^^^^^^^^^^^^^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
error: could not compile `jni` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `librespot v0.3.1`,
```

This is probably a problem of my Termux setup, but I found it curious that some JVM (?) software would be required to run librespot the rust library. (I'm now trying the librespot-java port)

Thanks for writing librespot!

**Comment by devgianlu** [comment]
You might also want to have a look at https://github.com/devgianlu/librespot-android

**Reply by nmlt** [reply]
That looks absolutely perfect! Thank you for your work with librespot!
